dev/src/reference.ts

```diff
@@ -1942,81 +1921,77 @@ export class CollectionReference extends Query {
         (other instanceof CollectionReference && super.isEqual(other)));
   }
 }
-/*!
- * Creates a new CollectionReference. Invoked by DocumentReference to avoid
- * invalid declaration order.
- *
- * @param {Firestore} firestore The Firestore Database client.
- * @param {ResourcePath} path The path of this collection.
- * @returns {CollectionReference}
- */
-function createCollectionReference(firestore, path): CollectionReference {
-  return new CollectionReference(firestore, path);
-}
 
-/*!
+/**
  * Validates the input string as a field order direction.
  *
- * @param {string=} str Order direction to validate.
- * @throws {Error} when the direction is invalid
+ * @private
+ * @param arg The argument name or argument index (for varargs methods).
+ * @param op Order direction to validate.
+ * @throws when the direction is invalid
  */
-export function validateFieldOrder(str: string): boolean {
-  if (!is.string(str) || !is.defined(directionOperators[str])) {
-    throw new Error('Order must be one of "asc" or "desc".');
-  }
-
-  return true;
+export function validateQueryOrder(arg: string|number, op: unknown): void {
+  validateEnumValue(arg, op, Object.keys(directionOperators), {optional: true});
```

Comment:
Per discussions with Gil, etc. during the push to GA, I think we want to unroll the breaking change we made here (and elsewhere).  And in fact we may want to relax the mobile SDK to match.